src/collision/contact_query.rs

```diff
@@ -201,7 +201,8 @@ pub fn contact_manifolds(
                         normal1,
                         normal2,
                         -contact.dist,
-                        0,
+                        PackedFeatureId::UNKNOWN,
```

Comment:
The `ContactData` constructor is very unreadable in its current state. I'd remove the `new` call and build it explicitly. Maybe add a `Default` so that you can do `..default()` for the three field that you leave at `0.0` / `default()`?

Reply:
There aren't really default contact points or normals that would make much sense, so I'm not sure about adding a `Default` impl. One option could be to at least remove the feature IDs from the constructor and make them default to `PackedFeatureId::UNKNOWN`, and to have a `with_feature_ids` method for when you need to specify them

Reply:
For now I just removed the feature IDs from the constructor and added a `with_feature_ids` method. I don't think the constructor is *too* unreadable, it's just two points, two normals, and a penetration depth. It would be even clearer if we used `Dir2`/`Dir3` for the normals though (I might change it to this in the future)